### PR TITLE
minigroups API

### PR DESCRIPTION
**Pull request by khodzha**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -13,13 +13,13 @@ tide = { version = "0.15.0", default_features = false, features = ["h1-server",
 async-std = { version = "1.8.0", features = ["attributes"] }
 async-trait = "0.1"
 chrono = { version = "0.4", features = ["serde"] }
-sqlx = { version = "0.5", features = ["offline", "postgres", "macros", "uuid", "chrono", "json", "bigdecimal", "runtime-async-std-rustls"] }
+sqlx = { version = "0.4", features = ["offline", "postgres", "macros", "uuid", "chrono", "json", "bigdecimal", "runtime-async-std-rustls"] }
```
Comment (feymartynov) [inline Cargo.toml:16]: Why downgrading?

File: src/app/api/v1/minigroup/mod.rs [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+struct MinigroupObject {
+    id: String,
+    real_time: RealTimeObject,
+}
+
+#[derive(Serialize)]
+struct RealTimeObject {
+    conference_room_id: Uuid,
+    event_room_id: Uuid,
+    fallback_uri: Option<String>,
```
Comment (feymartynov) [inline src/app/api/v1/minigroup/mod.rs:31]: There's no point falling back a minigroup so this seems redundand.

File: src/app/api/v1/minigroup/mod.rs [hunk 3]
```diff
@@ -0,0 +1,330 @@
...
+}
+async fn read_inner(req: Request<Arc<dyn AppContext>>) -> AppResult {
+    let account_id = fetch_token(&req).error(AppErrorKind::Unauthorized)?;
+    let state = req.state();
+
+    let minigroup = find_minigroup(&req)
+        .await
+        .error(AppErrorKind::WebinarNotFound)?;
+
+    let object = AuthzObject::new(&["minigroups", &minigroup.id().to_string()]).into();
```
Comment (feymartynov) [inline src/app/api/v1/minigroup/mod.rs:60]: Create a task plz for the tenant to add new authz object in mini-groups epic.

File: src/app/api/v1/minigroup/mod.rs [hunk 4]
```diff
@@ -0,0 +1,330 @@
...
+    let response = Response::builder(201).body(body).build();
+
+    Ok(response)
+}
+
+#[derive(Deserialize)]
+struct MinigroupUpdate {
+    #[serde(with = "crate::serde::ts_seconds_bound_tuple")]
+    time: BoundedDateTimeTuple,
+}
```
Comment (feymartynov) [inline src/app/api/v1/minigroup/mod.rs:229]: It maybe necessary to update the host.

File: Cargo.toml (hunk 1 above)
Comment (khodzha) [reply Cargo.toml:16]: using sqlx 0.5 will require sqlx updates in `svc-agent` and `svc-authn`

yet this upgrade requires specifying tls implementation for compiling `sqlx-rt` (dependency of `sqlx-core`) which requires pulling in a runtime (async-std in our case) in svc libs

this doesn't look like a proper thing to do thus this downgrade

the fix is expected to arrive in 0.6
https://github.com/launchbadge/sqlx/issues/996

File: src/app/api/v1/minigroup/mod.rs (hunk 2 above)
Comment (khodzha) [reply src/app/api/v1/minigroup/mod.rs]: removed

File: src/app/api/v1/minigroup/mod.rs (hunk 4 above)
Comment (khodzha) [reply src/app/api/v1/minigroup/mod.rs:228]: added

**Review by feymartynov — APPROVED** [review]

File: docs/src/minigroups/api.md [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+Request parameters:
+
+Attribute              | Type        | Optional | Description
+---------------------- | ----------- | -------- | -------------------------------------------------
+scope                  | string      |          | Scope
+audience               | string      |          | Audience
+time                   | [int, int]  | +        | Start and end
+tags                   | json object | +        | Arbitrary tags.
+host                   | string      |          | Host of the nightmare
```
Comment (feymartynov) [inline docs/src/minigroups/api.md:23]: 😈

Comment (feymartynov) [reply docs/src/minigroups/api.md:23]: `Host account id` would be a more useful description.

**Review by manifest — APPROVED** [review]

**Pull request merged**